### PR TITLE
let list creation use url params

### LINKED ISSUE #8245: Allow creation of lists via hand-crafted links of seed items
A link like the following (you may need to change the username in the url to your own) will pre-populate the add a list page with the specified items:

https://openlibrary.org/people/davidscotson/lists/add?name=The+Murderbot+Diaries&seeds=OL24747433W,OL17914663W,OL27481052W,OL19753589W,OL26794585W,OL24349461W,OL20735675W,OL20805971W,OL33402895W

This superficially looks the same as if you visited the "add a list" url and manually added the items. But on saving you get a 500 error if you use the url method.

On investigation, I believe this is due to the HTML form not specifying an "action" attribute, and so it tries to POST the form the to the full url, rather than just "https://openlibrary.org/people/davidscotson/lists/add".

Adding that shorter url to the <form> as an action attribute makes this work again.

### Details

- **Logged in (Y/N)?** Y

- **Environment (prod/dev/local)?** prod

### PR DESCRIPTION
**Pull request by davidscotson**
Closes #8245

small feature

### Technical

It feels slightly hacky to use ? as the action rather than the full url to clear out any exitsing url params, but not sure if that's just me.

### Testing

1. Go to the "create a list" UI to start creating an empty list.
2. Add the following to the URL `?name=Flatland&description=list-testing-desc&seeds=OL118388W`
3. Load that url and it should create a list with a title, description and book pre-populated.
4. Submitting the form should succeed in creating the list.

(I don't know much about how the debug mode works, but I believe the above should also work when in debug mode)

### CODE REVIEW

File: openlibrary/templates/type/list/edit.html [hunk 1]
```diff
@@ -85,7 +85,7 @@ <h1>$(_("Create a list") if new else _("Edit List"))</h1>
     </div>
 
 <div id="contentBody">
-    <form method="post" id="list-edit" class="olform" $:cond(query_param('debug'), 'action="?debug=true"')>
+	<form method="post" id="list-edit" class="olform" action="?$:cond(query_param('debug'), 'debug=true')">
```
Comment (jimchamp) [inline openlibrary/templates/type/list/edit.html:88]:
```suggestion
    <form method="post" id="list-edit" class="olform" action="?">
```

**Review by jimchamp — CHANGES_REQUESTED** [review]
Setting `action="?"` will make this work, but masks the underlying issue of `utils.unflatten` receiving data in an unexpected format.

I compared the differences between POSTing an add list form that was populated manually, and one that was populated by query parameters.  When logging the value of `web.input()` in the POST handler, the only difference between these types of requests is the value of `seeds`.

Manually filled form input:
`<Storage {'seeds--1--key': '/works/OL286811W', 'description': 'desc', 'name': 'list_name', 'seeds--0--key': '/books/OL4M', 'key': None, 'seeds': []}>`

Automatically filled form input:
`<Storage {'description': 'desc', 'name': 'list_name', 'seeds': ['/books/OL4M,/works/OL286811W'], 'seeds--1--key': '/works/OL286811W', 'seeds--0--key': '/books/OL4M', 'key': None}>`

I'll let you decide what to do, @davidscotson.  EIther use the tacky solution of setting action to `"?"`, or figure out why `seeds` is being populated differently when query params are used.  If you choose the latter, @cdrini will be a good resource to bounce questions off of, as he just created this feature recently.

**Comment by davidscotson** [comment]
Some minor progress on figuring out what's going on, urls like this:

http://localhost:8080/people/openlibrary/lists/add?seeds=OL5702375W,OL1W&seeds--0--key=OL4W&debug=true

Demonstrate that the list generation code generally doesn't like being given both seeds and seeds--0--key format parameters at the same time. I haven't quite traced it all the way through but something about this combination makes it try to treat a list like a dictionary, causing the error.

**Comment by davidscotson** [comment]
@jimchamp I think I got to the bottom of the issue. I've changed the PR to make a change to the unflatten util function that should avoid the issue, and shouldn't affect any existing usage, since they would have errored if they'd hit this codepath.

**Comment by jimchamp** [comment]
@cdrini to provide feedback re: `unflatten` function.

**Comment by davidscotson** [comment]
Worth noting that the infogami `unflatten` function which I guess this is inspired by, has a comment about a similar clash:

https://github.com/infogami/infogami/blob/3e9357c0184cd51921e198762c9f7d03b4589cf1/infogami/core/helpers.py#L50

**Pull request closed without merging**